subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/plugins/PrecompiledGroovyPluginsIntegrationTest.groovy

```diff
@@ -957,6 +958,55 @@ class PrecompiledGroovyPluginsIntegrationTest extends AbstractIntegrationSpec {
         succeeds(SAMPLE_TASK)
     }
 
+    def "should not allow precompiled plugin to conflict with core plugin"() {
+        given:
+        enablePrecompiledPluginsInBuildSrc()
+
+        file("buildSrc/src/main/groovy/plugins/java.gradle") << """
+            class TestTask extends DefaultTask {
+                @TaskAction
+                void run() {
+                    println 'from custom task'
```

Comment:
nitpick: we don't verify the output; we can just remove the printline.

Reply:
Task is not even needed for this test, removed